packages/client/src/components/app/blocks/TableBlock.svelte

```diff
@@ -249,7 +249,7 @@
             props={{
               dataSource,
               saveButtonLabel: sidePanelSaveLabel || "Save", //always show
-              deleteButtonLabel: deleteLabel, //respect config
+              deleteButtonLabel: deleteLabel == "" ? "" : "Delete",
```

Comment:
Does the non-strict equality get us anything an impilict truthiness check wouldn't here? Just checking as `==` always confuses me 🙃

Reply:
Yep, there are two things in play here. There's a backwards compatibility check at the top that looks for the old toggle boolean and ensures that it is taken into account first.

If the legacy value was explicitly `false` we change the label value to `""`. This implies then, on line 252, that a user decision to hide the button was made and so an empty string will now be used for the `Delete button` label, hiding it.

If no value was ever set for the `Delete button` label in the screen config, this value is explicitly `undefined` and considered to have never been set. In that scenario the default behaviour is to show the button with the `Delete` label.

Reply:
I had a look over this area and decided to refactor all the delete label code into its own function. It was difficult to follow and the line you highlighted is actually a mistake on my part. Cheers 👍